src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Extensions/ContextQuery/SyntaxTreeExtensions.cs

```diff
@@ -1844,6 +1865,13 @@ SyntaxKind.ForStatement or
                 return true;
             }
 
+            // scoped |
+            if ((token.IsKind(SyntaxKind.IdentifierToken) && token.Text == "scoped" && token.Parent.IsKind(SyntaxKind.IdentifierName) && token.Parent.Parent!.Kind() is SyntaxKind.VariableDeclaration or SyntaxKind.ExpressionStatement or SyntaxKind.IncompleteMember) ||
+                token.IsKind(SyntaxKind.ScopedKeyword) && token.Parent!.Kind() is SyntaxKind.ScopedType or SyntaxKind.IncompleteMember)
```

Comment:
&&'s and ||'s are kiling me. just break these into two cases.  and explain the different cases :)

Reply:
I found cases that are unhandled while addressing this comment, e.g:

```csharp
scoped $$

class C { }
```

and possibly cases that are either unnecessary or might not be covered by tests. I'll look again.